pkg/sql/sqlliveness/sqlliveness.go

```diff
@@ -20,12 +20,23 @@ import (
 	"encoding/hex"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/catpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
 	"github.com/cockroachdb/cockroach/pkg/util/encoding"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/metric"
 	"github.com/cockroachdb/errors"
 )
 
+// TypeDescriptor represents a stripped down version of catalog.TypeDescriptor.
+// We do this to avoid a circular dependency on pkg/sql/catalog, which imports
+// pkg/sql/sem/eval. The latter package imports pkg/sql/sqlliveness.
```

Comment:
This dependency of catalog on eval is a shameful wart which the SQL Schema team has meant to remove for a while now but still hasn't. Sorry about that!

In any case even if that obstacle weren't in the way, the dependency structure of your change bothers me a bit. I may not have the full picture here but it feels like you're pushing too much high-level logic into `SetRegionalData` when in fact it would be better off being more "dumb" and instead just take the `[]byte`s for the physical representation of the region, either nakedly or wrapped in some lightweight abstraction.

This could in turn originate from a new method you could add on `multiregion.RegionConfig` and which you call in the `server` package. This way no need to shadow `catalog.TypeDescriptor` at all, which is just as well.